### PR TITLE
Make DeploymentLauncher's stop command run in parallel

### PR DESCRIPTION
**Pull request by padhansell**
#### Description
Refactored the DeploymentLauncher's stop command to execute multiple DeleteDeployment commands simultaneously (whereas it previously used to block waiting for each deployment in serial). This dramatically decreases the time it takes to stop lots of deployments which is useful when working with large numbers of simplayers. 

#### Release note
DeploymentLauncher's "stop" command now runs in parallel in order to improve performance when working with large numbers of deployments.

#### Tests
Ran the stop command multiple times with both code flows (single deployment by name, and all deployments)

#### Documentation
Release note

### CODE REVIEW

File: SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/DeploymentLauncher/DeploymentLauncher.cs [hunk 1]
```diff
@@ -683,49 +684,61 @@ private static int StopDeployments(string[] args, bool useChinaPlatform)
...
-            var activeDeployments = ListLaunchedActiveDeployments(deploymentServiceClient, projectName);
-
-            foreach (var deployment in activeDeployments)
+            else
             {
-                var deploymentId = deployment.Id;
-                StopDeploymentById(deploymentServiceClient, projectName, deploymentId);
+                // Stop all active deployments launched by this launcher.
+                var activeDeployments = ListLaunchedActiveDeployments(deploymentServiceClient, projectName);
+                foreach( var deployment in activeDeployments)
```
Comment (improbable-valentyn) [inline SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/DeploymentLauncher/DeploymentLauncher.cs:699]:
```suggestion
                foreach (var deployment in activeDeployments)
```

File: SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/DeploymentLauncher/DeploymentLauncher.cs [hunk 2]
```diff
@@ -683,49 +684,61 @@ private static int StopDeployments(string[] args, bool useChinaPlatform)
...
-                });
+                Task.WaitAll(tasks.ToArray());
             }
-            catch (Grpc.Core.RpcException e)
+            catch
             {
-                if (e.Status.StatusCode == Grpc.Core.StatusCode.NotFound)
+                // Retrieve individual exceptions from AggregateException thrown by Task.WaitAll
+                var throwers = tasks.Where(task => task.Exception != null);
+                foreach(Task erroredTask in throwers)
```
Comment (improbable-valentyn) [inline SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/DeploymentLauncher/DeploymentLauncher.cs:722]:
```suggestion
                foreach (Task erroredTask in throwers)
```

File: SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/DeploymentLauncher/DeploymentLauncher.cs [hunk 3]
```diff
@@ -683,49 +684,61 @@ private static int StopDeployments(string[] args, bool useChinaPlatform)
...
                 {
-                    Console.WriteLine("<error:unknown-deployment>");
-                }
-                else
-                {
-                    throw;
+                    Exception inner = erroredTask.Exception.InnerException;
+                    Console.WriteLine($"Error while stopping deployment: {inner.Message}");
```
Comment (improbable-valentyn) [inline SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/DeploymentLauncher/DeploymentLauncher.cs:725]: Could you also log the deployment name that had the error?

**Review by improbable-valentyn — APPROVED** [review]
Looks good, please also add deployment ID in the error log.

File: SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/DeploymentLauncher/DeploymentLauncher.cs (hunk 3 above)
Comment (padhansell) [reply SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/DeploymentLauncher/DeploymentLauncher.cs]: Good point - done!

**Pull request merged**